File: classes/parte-7-paradigmas/110-orientado-a-objetos-clases-objetos-y-estado/implementaciones/javascript/main.mjs
import { readFileSync } from "node:fs";

class Contador {
  constructor() {
    this.cuenta = 0;
  }
  incrementar() {
    this.cuenta++;
  }
}

const n = parseInt(readFileSync(0, "utf8").trim(), 10);
const c = new Contador();
for (let i = 0; i < n; i++) c.incrementar();
console.log(`cuenta=${c.cuenta}`);

File: classes/parte-7-paradigmas/110-orientado-a-objetos-clases-objetos-y-estado/implementaciones/c/main.c
#include <stdio.h>

struct Contador {
    long cuenta;
};

void incrementar(struct Contador *c) {
    c->cuenta++;
}

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    struct Contador c = {0};
    for (long i = 0; i < n; i++) incrementar(&c);
    printf("cuenta=%ld\n", c.cuenta);
    return 0;
}
